### PR TITLE
Move to go modules 

### PR DESCRIPTION
**Pull request by katiewasnothere**
This PR removes dependency on the deprecated vendor.conf for handling dependencies by moving to go modules. Additional changes required include: 

- remove any references to k8s.io/kubernetes to avoid go modules complaining
- change docker imports to containerd matching packages
- update calls to net plugin to take a context variable to match the upstream api
- add new calls added in upstream cri-api to our cri service 
- Use the correct event plugin when registering our containerd event service 
- make sure all necessary code from containerd is pulled in when vendoring 

Note: cri's v1alpha2 api is depracated as well. Since there is code in containerd relying on the v1 api instead, this PR will cause a build break. A followup PR will be made based on some of the changes in https://github.com/kevpar/cri/pull/29 to remedy this. 

**PS: I recommend reviewing this by looking at the individual `commits` instead of via the `files changed` tab since this PR contains a lot of file changes from vendoring**

### CODE REVIEW

**Comment by kevpar** [comment]
> - remove any references to k8s.io/kubernetes to avoid go modules complaining
> - change docker imports to containerd matching packages
> - update code to remove custom signal parsing since that's been removed in upstream
> - update calls to net plugin to take a context variable to match the upstream api
> - replace genproto and grpc imports to avoid panic in ttrpc module. The same is done in https://github.com/microsoft/hcsshim/pull/681.

I'm curious if these changes were strictly required as part of moving to modules, or if they are just other good-to-have things. If the latter, I think it would be easier to review if we split these into multiple PRs, and leave this one just for the module conversion.

**Comment by katiewasnothere** [comment]
> > * remove any references to k8s.io/kubernetes to avoid go modules complaining
> > * change docker imports to containerd matching packages
> > * update code to remove custom signal parsing since that's been removed in upstream
> > * update calls to net plugin to take a context variable to match the upstream api
> > * replace genproto and grpc imports to avoid panic in ttrpc module. The same is done in [use ttrpc for event publishing to containerd microsoft/hcsshim#681](https://github.com/microsoft/hcsshim/pull/681).
> 
> I'm curious if these changes were strictly required as part of moving to modules, or if they are just other good-to-have things. If the latter, I think it would be easier to review if we split these into multiple PRs, and leave this one just for the module conversion.

Unfortunately, all of these changes were required to successfully build CRI with modules containerd 1.6. As we discussed offline, I'm going to try to move around the commits to make the review process easier since this is a lot of file changes :)

**Comment by kevpar** [comment]
> > > * remove any references to k8s.io/kubernetes to avoid go modules complaining
> > > * change docker imports to containerd matching packages
> > > * update code to remove custom signal parsing since that's been removed in upstream
> > > * update calls to net plugin to take a context variable to match the upstream api
> > > * replace genproto and grpc imports to avoid panic in ttrpc module. The same is done in [use ttrpc for event publishing to containerd microsoft/hcsshim#681](https://github.com/microsoft/hcsshim/pull/681).
> > 
> > 
> > I'm curious if these changes were strictly required as part of moving to modules, or if they are just other good-to-have things. If the latter, I think it would be easier to review if we split these into multiple PRs, and leave this one just for the module conversion.
> 
> Unfortunately, all of these changes were required to successfully build CRI with modules containerd 1.6. As we discussed offline, I'm going to try to move around the commits to make the review process easier since this is a lot of file changes :)

It would be great to have some more information on what makes these required. For instance it's not intuitively obvious to me why changes in upstream would impact a move from vndr to modules, but I'm probably missing something.

One other note: I noticed this PR is also changing to `fork/release/1.6` for the containerd fork. Can we do that in a separate PR and keep this one strictly vndr -> modules?

**Comment by katiewasnothere** [comment]
> One other note: I noticed this PR is also changing to `fork/release/1.6` for the containerd fork. Can we do that in a separate PR and keep this one strictly vndr -> modules?

k8s.io/kubernetes is not intended to be vendored in with go modules. As a result, we need to replace any references to those packages. A lot of the code that we previously used from k8s.io/kubernetes was refactored in various ways and some of it landed in containerd. Since those code refactors are more recent, we're forced to move to release 1.5 or 1.6 to reference them. This accounts for all the code changes beyond just moving from vndr -> go modules. 

> It would be great to have some more information on what makes these required. 

I've updated the PR with more descriptive commits in the hopes that it helps with reviewing and also with determining why some changes are needed. 

> For instance it's not intuitively obvious to me why changes in upstream would impact a move from vndr to modules, but I'm probably missing something.

There were some upstream changes to the cri api and cni api, resulting in us needing changes here to build. Let me know if I can elaborate more on this.

**Comment by katiewasnothere** [comment]
I've also updated the PR description to better reflect the commits here

File: pkg/server/sandbox_run.go [hunk 1]
```diff
@@ -69,7 +69,8 @@ func (c *criService) setupPod(ctx context.Context, id string, path string, confi
 	}
 
 	labels := getPodCNILabels(id, config)
-	result, err := c.netPlugin.Setup(id,
+	result, err := c.netPlugin.Setup(ctx,
```
Comment (helsaawy) [inline pkg/server/sandbox_run.go:72]: did the `netPlugin` interface change when we upgraded to a new version of containerd?

File: pkg/server/sandbox_run_unix.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		defer func() {
 			if retErr != nil {
 				// Teardown network if an error is returned.
-				if err := c.teardownPod(id, sandbox.NetNSPath, config); err != nil {
+				if err := c.teardownPod(ctx, id, sandbox.NetNSPath, config); err != nil {
```
Comment (helsaawy) [inline pkg/server/sandbox_run_unix.go:138]: do we have to keep the `*_unix` files, or can we just delete them all and save ourselves the trouble?

**Review by helsaawy — APPROVED** [review]

File: pkg/server/sandbox_run.go (hunk 1 above)
Comment (katiewasnothere) [reply pkg/server/sandbox_run.go:72]: It was updated in go-cni. We could force use a previous version of the package if preferred.

File: pkg/server/sandbox_run_unix.go (hunk 2 above)
Comment (katiewasnothere) [reply pkg/server/sandbox_run_unix.go:138]: I think we should keep them for now while we decide what we should do with our cri fork

File: pkg/server/sandbox_run.go (hunk 1 above)
Comment (helsaawy) [reply pkg/server/sandbox_run.go:72]: no preference, just curious why it changed.

**Comment by katiewasnothere** [comment]
Rebased to accommodate changes from https://github.com/kevpar/cri/pull/36 and https://github.com/kevpar/cri/pull/37

File: cmd/containerd/builtins_windows.go [hunk 3]
```diff
@@ -24,4 +24,5 @@ import (
 	_ "github.com/containerd/containerd/runtime/v2"
 	_ "github.com/containerd/containerd/snapshots/lcow"
 	_ "github.com/containerd/containerd/snapshots/wcow"
+	_ "github.com/containerd/containerd/snapshots/windows"
```
Comment (kevpar) [inline cmd/containerd/builtins_windows.go]: Why is this needed? It's referenced from `github.com/containerd/containerd/snapshots/wcow`, so should already be pulled in?

**Comment by kevpar** [comment]
After this change, `windows_port` will only work with `fork/release/1.6` from our fork or newer, due to packages we depend on that were added. Is that right?

File: cmd/containerd/builtins_windows.go (hunk 3 above)
Comment (katiewasnothere) [reply cmd/containerd/builtins_windows.go]: Oh weird. When I was experimenting, the files for "github.com/containerd/containerd/snapshots/windows" weren't getting pulled in which caused build issues in "github.com/containerd/containerd/snapshots/wcow".

Comment (kevpar) [reply cmd/containerd/builtins_windows.go]: Interesting. Generally this `import _` pattern is only needed when we want to get the side effects from an `init` routine but nothing actually references the package anywhere.

**Comment by katiewasnothere** [comment]
> After this change, `windows_port` will only work with `fork/release/1.6` from our fork or newer, due to packages we depend on that were added. Is that right?

That is correct. I've made a new branch in this repo here https://github.com/kevpar/cri/tree/fork/windows_port/1.4 for working with `fork/release/1.4` in our containerd fork if we need it in the future.

File: cmd/containerd/builtins_windows.go (hunk 3 above)
Comment (katiewasnothere) [reply cmd/containerd/builtins_windows.go]: Ah, looks like it is indeed not needed. Not sure how I ran into an issue with that haha I'll remove it and rebase the PR.

Comment (katiewasnothere) [reply cmd/containerd/builtins_windows.go]: Removed.
